xstream/src/test/com/thoughtworks/xstream/io/naming/StaticNameCoderTest.java

```diff
@@ -0,0 +1,72 @@
+package com.thoughtworks.xstream.io.naming;
+
+import com.thoughtworks.xstream.io.naming.StaticNameCoder;
+import junit.framework.TestCase;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class StaticNameCoderTest extends TestCase {
+
+  public void testDecodeAttribute() {
+    final StaticNameCoder staticNameCoder = new StaticNameCoder(new HashMap<String, String>(), new HashMap<String, String>());
+    assertEquals("\"  ", staticNameCoder.decodeAttribute("\"  "));
+  }
+
+  public void testDecodeAttribute2() {
+    // Arrange
+    final HashMap<String, String> hashMap = new HashMap<String, String>();
+    hashMap.put("", null);
+    final StaticNameCoder staticNameCoder = new StaticNameCoder(new HashMap<String, String>(), hashMap);
+    // Act & Assert result
+    assertEquals("", staticNameCoder.decodeAttribute(null));
+  }
+
+  public void testDecodeNode() {
+    final StaticNameCoder staticNameCoder = new StaticNameCoder(new HashMap<String, String>(), new HashMap<String, String>());
+    assertEquals("   ", staticNameCoder.decodeNode("   "));
+  }
+
+  public void testDecodeNode2() {
+    // Arrange
+    final HashMap<String, String> java2Node = new HashMap<String, String>();
+    final HashMap<String, String> java2Attribute = new HashMap<String, String>();
+    java2Node.put("", null);
+    java2Attribute.put(null, null);
+    final StaticNameCoder staticNameCoder = new StaticNameCoder(java2Node, java2Attribute);
+    // Act & Assert result
+    assertEquals("", staticNameCoder.decodeNode(null));
+  }
+
+  public void testEncodeAttribute() {
+    final StaticNameCoder staticNameCoder = new StaticNameCoder(new HashMap<String, String>(), null);
+    assertEquals("$$$$$$$$$", staticNameCoder.encodeAttribute("$$$$$$$$$"));
+  }
+
+  public void testEncodeAttribute2() {
+    // Arrange
+    final HashMap<String, String> java2Node = new HashMap<String, String>();
```

Comment:
In all HashMap assignments, the constructor call doesn't need explicit type parameters:
`final HashMap<String, String> java2Node = new HashMap<>();`

Reply:
Fixed.